packages/@aws-cdk/cloud-assembly-schema/lib/cloud-assembly/artifact-schema.ts

```diff
@@ -91,6 +91,13 @@ export interface AwsCloudFormationStackProperties {
    * @default - false
    */
   readonly validateOnSynth?: boolean;
+
+  /**
+   * The role to use to look up values from the target AWS account during synthesis
+   *
+   * @default - None
+   */
+  readonly lookupRoleArn?: string;
```

Comment:
Technically, we also need the ability to attach a minimum bootstrap version to this role ARN. The lookup role was introduced in a bootstrap stack version 7 (or something), but nothing in the CLI so far has forced users to actually update to that bootstrap version, and if they haven't it won't work.

Can we turn it into the following:

```
interface CloudFormationStackArtifact {
  readonly lookupRole?: BootstrapRole;
}

interface BootstrapRole {
  readonly arn: string;
  readonly requiresBootstrapStackVersion?: number;
}
```

## But why not bump the requiresBootstrapStackVersion version on the stack itself?

That would require *everyone* to update their bootstrap stack, whether or not they will be using this new feature. That is not a good user experience. We should be limiting the bootstrap stack requirements to individual features that a user is actually going to use (otherwise people *will* complain about breaking changes!)

(Don't forget to use `bootstrapStackVersionSsmParameter` from the stack object, this is necessary in case people rename their bootstrap stack-- I'm even okay folding that into the `BootstrapRole` struct if the code in the CLI becomes awkward to write otherwise)

## This change by itself should probably go into a separate PR

I appreciate you diving onto this, but it's probably a change worthy of its own PR, for ease of reviewing. Can you please split it out?

Reply:
Ok I reverted all changes to the cloud assembly schema.

Reply:
@rix0rrr @skinny85 split this into a separate PR https://github.com/aws/aws-cdk/pull/18277